test/e2e/cluster_authorized_cidrs_connectivity.go

```diff
@@ -0,0 +1,255 @@
+// Copyright 2025 Microsoft Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package e2e
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/base64"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+
+	"github.com/Azure/ARO-HCP/test/util/framework"
+	"github.com/Azure/ARO-HCP/test/util/labels"
+)
+
+var _ = Describe("Authorized CIDRs", func() {
+	Context("Connectivity", func() {
+		It("should allow API access only from authorized VM IP",
+			labels.RequireNothing,
+			labels.Critical,
+			labels.Positive,
+			func(ctx context.Context) {
+				const (
+					clusterName                      = "cidr-connectivity-test"
+					customerNetworkSecurityGroupName = "customer-nsg-name"
+					customerVnetName                 = "customer-vnet-name"
+					customerVnetSubnetName           = "customer-vnet-subnet1"
+					openshiftControlPlaneVersionId   = "4.19"
+				)
+
+				tc := framework.NewTestContext()
+
+				By("creating a resource group")
+				resourceGroup, err := tc.NewResourceGroup(ctx, "e2e-cidr-connectivity", tc.Location())
+				Expect(err).NotTo(HaveOccurred())
+
+				By("creating cluster parameters")
+				clusterParams := framework.NewDefaultClusterParams()
+				clusterParams.ClusterName = clusterName
+				managedResourceGroupName := framework.SuffixName(*resourceGroup.Name, "-managed", 64)
+				clusterParams.ManagedResourceGroupName = managedResourceGroupName
+				clusterParams.OpenshiftVersionId = openshiftControlPlaneVersionId
+
+				By("creating customer resources")
+				clusterParams, err = tc.CreateClusterCustomerResources(ctx,
+					resourceGroup,
+					clusterParams,
+					map[string]interface{}{
+						"persistTagValue":        false,
+						"customerNsgName":        customerNetworkSecurityGroupName,
+						"customerVnetName":       customerVnetName,
+						"customerVnetSubnetName": customerVnetSubnetName,
+					},
+					TestArtifactsFS,
+				)
+				Expect(err).NotTo(HaveOccurred())
+
+				By("generating SSH key pair for VM")
+				sshPublicKey, _, err := framework.GenerateSSHKeyPair()
+				Expect(err).NotTo(HaveOccurred())
+
+				By("deploying test VM")
```

Comment:
Do this in parallel with the cluster creation.

Reply:
Does that prevent the creation of the cluster itself and VM ARM resources being parallelized? As far as I understand the VM deployment, its only dependencies are on the vnet/subnet (which should be put down by CreateClusterCustomerResources). I could very well be off-base or missing something, though.

Reply:
correction - as of now the cluster creation depends on the VM (not the reverse), as the VM Bicep template deploys the publicIpAddresses resource that we need to reference in both VM and cluster creation. 

A follow-up fix could be to split this IP address resource into its own deployment, but I don't think we need to block on that, unless we already are observing this VM creation taking the full ~30 minutes we timeout on.

Reply:
> Does that prevent the creation of the cluster itself and VM ARM resources being parallelized?

Yes. The  underlying fw infrastructure will block the traffic otherwise.